crypto/conf/conf_mod.c

```diff
@@ -152,6 +155,25 @@ int CONF_modules_load_file(const char *filename, const char *appname,
     return ret;
 }
 
+int CONF_modules_load_file(const char *filename,
+                           const char *appname, unsigned long flags)
+{
+    char *file = (char *)filename;
+    int ret = 0;
+
+    if (file == NULL) {
+        /* If we specify a default configuration to load try to load it */
+        file = CONF_get1_default_config_file();
+        /* dont fail if there is no default though */
+        if (file == NULL)
```

Comment:
Why do we need to do this? It seems to change the old behaviour. Previously, AFAICT, we only succeeded if the CONF_MFLAGS_IGNORE_RETURN_CODES flag was set. Now we succeed unconditionally. Why not just defer all of this to CONF_modules_load_file_with_libctx, and if the flag is set it succeeds?

Reply:
This is related to your other question below ...
OPENSSL_init_crypto(OPENSSL_INIT_LOAD_CONFIG) is needed to be called once..
which then calls openssl_config_int() which then calls CONF_modules_load_file().. So I am trying to deal with this case..
So maybe I need instead to do this code as a seperate copy that gets called inside the CONF_modules_load_file().

Reply:
Note that the way I have changed the code - this call it will only load the default configuration if the environment variable OPENSSL_CONF is set. This means that all existing test code works how it did before without breaking (since OPENSSL_CONF is used by quite a few tests).

Reply:
Sorry, I'm still not getting it...

> calls openssl_config_int() which then calls CONF_modules_load_file().

But `openssl_config_int()` calls `CONF_modules_load_file()` with the flags `DEFAULT_CONF_MFLAGS` which includes `CONF_MFLAGS_IGNORE_RETURN_CODES`. So this would mean `CONF_modules_load_file_with_libctx` shouldn't fail if there isn't a default config file anyway?

Reply:
See comment about deadlock above, its related to the same problem..

Reply:
This lazy init stuff is just plain nasty :)

Reply:
Ok so this function can now just call the libctx variation now that the dependancies are fixed up.